### PR TITLE
chore: gas savings audit changes

### PR DESCRIPTION
**Pull request by brandonchuah**
Audit report: https://notes.watchpug.com/p/18cb3b30088GDV1y#n_3

[[WP-G2] Merging the events LogUseGelato1Balance and ExecSuccess1Balance can save gas.
](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_4)

Not implemented as we have to follow 1Balance standard (check if 1Balance team)

[[WP-G3] The taskModules design may be over-engineering (for the current application).
](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_9)[[WP-G5] Redundant SLOAD taskModuleAddresses.
](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_11)

My view is that we will keep the modules and `exec1Balance` as it is to avoid a huge refactor. However, `execBypassModule` will be the preferred execution method and we will transition into using this.

[[WP-I6] Idea: If we remove _execAddress from calldata and instead calculate it based on _taskCreator or read it from storage, it might be cheaper on L2.](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_5)

To achieve this, we have to determine the dedicated msg sender via a call to `OpsProxyFactory`. `OpsProxyFactory` has to be stored as storage (1 read) / queried dynamically from the OpsProxy Module contract (which will require 1 state read & 1 call). 

I prefer to keep `_execAddress` so that `execBypassModule` can be used to execute tasks that do not enable `PROXY` module.

[[WP-O7] _call() and _delegateCall() will discard custom error information and instead revert with "NoErrorSelector".
](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_14)[[WP-G8] Unnecessary internal function calls
](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_15)[[WP-G9] Add OpsProxyFactory.proxyOf(address _account) storage getter to save unnecessary computation and gas waste from SLOAD.
](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_18)[[WP-I10] Idea: Make the Automate contract inherit OpsProxyFactory (Automate is OpsProxyFactory) can reducing external calls.
](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_18)

I see these as major changes and will also alter the addresses of user's dedicated msg sender which we want to avoid as of now.

### CODE REVIEW

File: contracts/opsProxy/OpsProxy.sol [hunk 1]
```diff
@@ -63,7 +63,15 @@ contract OpsProxy is Proxied, IOpsProxy {
         bytes calldata _data,
         uint256 _value
     ) private {
-        _call(_target, _data, _value, true, "OpsProxy.executeCall: ");
+        (, bytes memory returnData) = _call(
```
Comment (goums) [inline contracts/opsProxy/OpsProxy.sol:66]: Isn't that modifying the bytecode of the OpsProxy?

**Comment by goums** [comment]
> [](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_3)
> 
> [[WP-G2] Merging the events LogUseGelato1Balance and ExecSuccess1Balance can save gas. ](https://notes.watchpug.com/p/18cb3b30088GDV1y#n_4)
> 
> Not implemented as we have to follow 1Balance standard (check if 1Balance team)

I think it is feasible since we are already changing the event from the previous version anyway

File: contracts/opsProxy/OpsProxy.sol (hunk 1 above)
Comment (brandonchuah) [reply contracts/opsProxy/OpsProxy.sol:66]: Yea i am reverting the changes made in #123  which will keep OpsProxy untouched 😅

**Review by goums — APPROVED** [review]

File: contracts/interfaces/IAutomate.sol [hunk 2]
```diff
@@ -80,7 +80,7 @@ interface IAutomate is IGelato1Balance {
      * @param revertOnFailure To revert or not if call to execAddress fails. (Used for off-chain simulations)
      * @param singleExec If the task is a SingleExec task. If true, task will be cancelled after execution.
      */
-    function exec1BalanceSimple(
+    function execBypassModule(
```
Comment (jack-the-pug) [inline contracts/interfaces/IAutomate.sol:83]: Missing `execBypassModuleSyncFee()`

**Pull request merged**